code/implementation/host-stats/src/main/java/io/cattle/platform/host/stats/api/ContainerStatsLinkHandler.java

```diff
@@ -0,0 +1,116 @@
+package io.cattle.platform.host.stats.api;
+
+import io.cattle.platform.api.link.LinkHandler;
+import io.cattle.platform.core.constants.CommonStatesConstants;
+import io.cattle.platform.core.constants.HostConstants;
+import io.cattle.platform.core.constants.InstanceConstants;
+import io.cattle.platform.core.model.Host;
+import io.cattle.platform.core.model.Instance;
+import io.cattle.platform.docker.util.DockerUtils;
+import io.cattle.platform.host.model.HostApiAccess;
+import io.cattle.platform.host.service.HostApiService;
+import io.cattle.platform.host.stats.utils.StatsConstants;
+import io.cattle.platform.object.ObjectManager;
+import io.github.ibuildthecloud.gdapi.context.ApiContext;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+public class ContainerStatsLinkHandler implements LinkHandler {
+
+    HostApiService hostApiService;
+    ObjectManager objectManager;
+
+    @Override
+    public String[] getTypes() {
+        return new String[] { InstanceConstants.TYPE_CONTAINER, HostConstants.TYPE };
+    }
+
+    @Override
+    public boolean handles(String type, String id, String link, ApiRequest request) {
+        return StatsConstants.CONTAINER_STATS.equalsIgnoreCase(link);
+    }
+
+    @Override
+    public Object link(String name, Object obj, ApiRequest request) throws IOException {
+        Host host = null;
+        Instance instance = null;
+
+        if (obj instanceof Instance) {
+            instance = (Instance) obj;
+            host = DockerUtils.getHostFromContainer(objectManager, instance);
+        } else if (obj instanceof Host) {
+            host = (Host) obj;
+        }
+
+        if (host == null) {
+            return null;
```

Comment:
I've pointed out other places where this causes a 404, but I'm ok with it in this case because I can't imagine a valid scenario where the host would be null here.

We could also possibly just raise an `IllegalStateException` instead.